modules/common_log_format.lua

```diff
@@ -420,10 +419,204 @@ function build_apache_grammar(log_format)
     return l.Ct(grammar)
 end
 
+local hap_wrap_string_token = l.P'"'
+local hap_wrap_string_token_last = nil
+local hap_wrap_string_token_global = nil
+
+local function set_token_mod(t)
+    local modify = string.sub(t, 1, 1)
+    local modtype = string.sub(t, -1)
+    if modtype == "Q" then
+        if modify == "-" then
+            hap_wrap_string_token_last = {false, hap_wrap_string_token}
+        elseif modify == "+" then
+            hap_wrap_string_token_last = {true, hap_wrap_string_token}
+
+        end
+    else 
+        error("modtype: "..modtype.." not supported yet")
+    end
+end
+
+local function capture_split(x)
+    local sep = l.P"|"
+    local elem = l.C((1 - sep)^0)
+    local m = l.Ct(elem * (sep * elem)^0)
+    local x = m:match(x)
+    if x == nil then x = {} end
+    return x
+end
+
+local proxyname = (1 - l.space - (l.S"()+!@#$%^&*+=/\{}[]<>,"))^1
```

Comment:
i had to remove the `<>` in order to match `<NOSRV>` in the haproxy logs when there are no backends available.